lightyear/src/connection/steam/mod.rs

```diff
@@ -3,13 +3,7 @@ use steamworks::networking_types::{NetworkingConfigEntry, NetworkingConfigValue}
 
 pub(crate) mod client;
 pub(crate) mod server;
-
-// NOTE: it looks like there's SingleClient can actually be called on multiple threads
-// - https://partner.steamgames.com/doc/api/steam_api#SteamAPI_RunCallbacks
-pub(crate) struct SingleClientThreadSafe(steamworks::SingleClient);
```

Comment:
So do you think it would be possible to use this as a wrapper to the user's provided `SteamworksClient` to avoid the `Arc<RwLock>`?